### PR TITLE
Added QMK support for GSKT-00

### PR DESCRIPTION
**Pull request by nachie**
Add QMK support for  GSKT-00 ~~PaladinSplit, PaladinTKL and PaladinPad pcbs~~

### CODE REVIEW

File: keyboards/paladinsplit/rules.mk [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+OPT_DEFS += -DINTERRUPT_CONTROL_ENDPOINT
+
+
+# Boot Section Size in *bytes*
+#   Teensy halfKay   512
+#   Teensy++ halfKay 1024
+#   Atmel DFU loader 4096
+#   LUFA bootloader  4096
+#   USBaspLoader     2048
+OPT_DEFS += -DBOOTLOADER_SIZE=4096
```
Comment (drashna) [inline keyboards/paladinsplit/rules.mk]: This isn't needed, since the bootloader is defined.

File: keyboards/paladintkl/keymaps/default/keymap.c [hunk 2]
```diff
@@ -0,0 +1,105 @@
+#include QMK_KEYBOARD_H
+
+const uint16_t PROGMEM fn_actions[] = {};
```
Comment (drashna) [inline keyboards/paladintkl/keymaps/default/keymap.c]: Could you remove the `fn_actions` block?  All of it's features have been added as keycodes, and been deprecated in favor of `process_record_user.

File: keyboards/paladintkl/keymaps/default/keymap.c [hunk 3]
```diff
@@ -0,0 +1,105 @@
...
+        KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,
+        KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,
+        KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_BTN1, KC_BTN2, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,
+        KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,
+        KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,
+        KC_TRNS, KC_TRNS, KC_TRNS, KC_BTN3, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS)
+
+};
+
+const macro_t *action_get_macro(keyrecord_t *record, uint8_t id, uint8_t opt) {
```
Comment (drashna) [inline keyboards/paladintkl/keymaps/default/keymap.c]: Could you remove the `action_get_macro` function?  It's been deprecated in favor of `process_record_user`.

File: keyboards/paladintkl/readme.md [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+Make example for this keyboard (after setting up your build environment):
+
+    make paladintkl:default
+
+Or to make and flash:
+
+    make paladintkl:default:dfu
+
+
+See [build environment setup](https://docs.qmk.fm/build_environment_setup.html) then the [make instructions](https://docs.qmk.fm/make_instructions.html) for more information.
```
Comment (drashna) [inline keyboards/paladintkl/readme.md]: Could you replace this link block with this, instead: 

```markdown
See the [build environment setup](https://docs.qmk.fm/#/getting_started_build_tools) and the [make instructions](https://docs.qmk.fm/#/getting_started_make_guide) for more information. Brand new to QMK? Start with our [Complete Newbs Guide](https://docs.qmk.fm/#/newbs).
```

File: keyboards/paladintkl/rules.mk [hunk 5]
```diff
@@ -0,0 +1,62 @@
...
+#     If no clock division is performed on the input clock inside the AVR (via the
+#     CPU clock adjust registers or the clock division fuses), this will be equal to F_CPU.
+F_USB = $(F_CPU)
+
+# Interrupt driven control endpoint task(+60)
+OPT_DEFS += -DINTERRUPT_CONTROL_ENDPOINT
+
+
+# Boot Section Size in *bytes*
+OPT_DEFS += -DBOOTLOADER_SIZE=4096
```
Comment (drashna) [inline keyboards/paladintkl/rules.mk]: This isn't needed, since the bootloader is defined.

File: keyboards/paladintkl/rules.mk [hunk 6]
```diff
@@ -0,0 +1,62 @@
...
+
+BOOTLOADER = atmel-dfu
+
+# Build Options
+#   comment out to disable the options.
+#
+BOOTMAGIC_ENABLE ?= yes	# Virtual DIP switch configuration(+1000)
```
Comment (drashna) [inline keyboards/paladintkl/rules.mk]: Could you remove the question marks

File: keyboards/paladintkl/rules.mk [hunk 7]
```diff
@@ -0,0 +1,62 @@
...
+SLEEP_LED_ENABLE ?= no  # Breathing sleep LED during USB suspend
+NKRO_ENABLE ?= yes		# USB Nkey Rollover - if this doesn't work, see here: https://github.com/tmk/tmk_keyboard/wiki/FAQ#nkro-doesnt-work
+BACKLIGHT_ENABLE ?= no  # Enable keyboard backlight functionality
+AUDIO_ENABLE ?= no
+RGBLIGHT_ENABLE ?= yes
+#PS2_MOUSE_ENABLE ?= yes
+#PS2_USE_USART ?= yes
+#PS2_USE_INT ?= yes
+
+LAYOUTS = 60_ansi
```
Comment (drashna) [inline keyboards/paladintkl/rules.mk]: If you don't have a `LAYOUT_60_ansi` in your keyboard's h file, you shouldn't use this layout. 

If you could fix either this or add/change the layout?

**Comment by nachie** [comment]
Removed the paladin boards from this PR, will submit another one with them in a different folder as suggested. There will be a group buy for these soon, so will submit once that is closer.

For the gskt00, there already was a private GB, and more than 60 pcbs are being produced, so I think that would qualify for the production.

Will work through rest of the comments :)

File: layouts/default/60_iso_tsangan/layout.json [hunk 8]
```diff
@@ -0,0 +1,5 @@
+[{a:7},"","","","","","","","","","","","","",{w:2},""],
+[{w:1.5},"","","","","","","","","","","","","",{x:0.25,w:1.25,h:2,w2:1.5,h2:1,x2:-0.25},""],
+[{w:1.75},"","","","","","","","","","","","",""],
+[{w:1.25},"","","","","","","","","","","","",{w:2.75},""],
+[{w:1.5},"",{w:1},"",{w:1.5},"",{w:7},"",{w:1.5},"",{w:1},"",{w:15},""]
```
Comment (noroadsleft) [inline layouts/default/60_iso_tsangan/layout.json]: Your last key on this row is 15u wide instead of 1.5u. :smile:

Comment (nachie) [reply layouts/default/60_iso_tsangan/layout.json]: Damn! Fixing that...

File: keyboards/gskt00/gskt00.h [hunk 9]
```diff
@@ -0,0 +1,57 @@
...
+    { K30, K31, K32, K33, K34, K35, K36, K37 }, \
+    { K40, K41, K42, K43, K44, K45, K46, K47 }, \
+    { K50, K51, K52, K53, K54, K55, K56, K57 }, \
+    { K60, K61, K62, K63, K64, K65, K66, K67 }, \
+    { KC_NO, K71, K72, K73, K74, K75, K76, K77 } \
+}
+
+#define LAYOUT_60_iso_tsangan( \
+    K36, K37, K46, K47, K56, K57, K66, K67, K76, K77, K06, K07, K17,      K27, \
+    K34, K35, K44, K45, K54, K55, K64, K65, K75, K05, K15, K16, K25, K24, \
```
Comment (noroadsleft) [inline keyboards/gskt00/gskt00.h]: `K24` needs to be moved down to between `K14` and `K23` for `LAYOUT_60_iso_tsangan` to work properly.

File: keyboards/gskt00/info.json [hunk 10]
```diff
@@ -0,0 +1,16 @@
...
..."y":1}, {"label":"Enter", "x":13.75, "y":1, "w":1.25, "h":2}, {"label":"Caps Lock", "x":0, "y":2, "w":1.75}, {"label":"A", "x":1.75, "y":2}, {"label":"S", "x":2.75, "y":2}, {"label":"D", "x":3.75, "y":2}, {"label":"F", "x":4.75, "y":2}, {"label":"G", "x":5.75, "y":2}, {"label":"H", "x":6.75, "y":2}, {"label":"J", "x":7.75, "y":2}, {"label":"K", "x":8.75, "y":2}, {"label":"L", "x":9.75, "y":2}, {"label":":", "x":10.75, "y":2}, {"label":"@", "x":11.75, "y":2}, {"label":"~", "x":12.75, "y":2}, {"label":"Shift", "x":0, "y":3, "w":1.25}, {"label":"|", "x":1.25, "y":3}, {"label":"Z", "x":2.25, "y":3}, {"label":"X", "x":3.25, "y":3}, {"label":"C", "x":4.25, "y":3}, {"label":"V", "x":5.25, "y":3}, {"label":"B", "x":6.25, "y":3}, {"label":"N", "x":7.25, "y":3}, {"label":"M", "x":8.25, "y":3}, {"label":"<", "x":9.25, "y":3}, {"label":">", "x":10.25, "y":3}, {"label":"?", "x":11.25, "y":3}, {"label":"Shift", "x":12.25, "y":3, "w":2.75}, {"label":"Ctrl", "x":0, "y":4, "w":1.5}, {"label":"Win", "x":1.5, "y":4}, {"label":"Alt", "x":2.5, "y":4, "w":1.5}, {"x":4, "y":4, "w":7}, {"label":"AltGr", "x":11, "y":4, "w":1.5}, {"label":"Menu", "x":12.5, "y":4}, {"label":"Ctrl", "x":13.5, "y":4, "w":1.5}]
```
Comment (noroadsleft) [inline keyboards/gskt00/info.json]: Related to the `K24` move:

Please change this line to the following:

```
            "layout": [{"label":"\u00ac", "x":0, "y":0}, {"label":"!", "x":1, "y":0}, {"label":"\"", "x":2, "y":0}, {"label":"\u00a3", "x":3, "y":0}, {"label":"$", "x":4, "y":0}, {"label":"%", "x":5, "y":0}, {"label":"^", "x":6, "y":0}, {"label":"&", "x":7, "y":0}, {"label":"*", "x":8, "y":0}, {"label":"(", "x":9, "y":0}, {"label":")", "x":10, "y":0}, {"label":"_", "x":11, "y":0}, {"label":"+", "x":12, "y":0}, {"label":"Backspace", "x":13, "y":0, "w":2}, {"label":"Tab", "x":0, "y":1, "w":1.5}, {"label":"Q", "x":1.5, "y":1}, {"label":"W", "x":2.5, "y":1}, {"label":"E", "x":3.5, "y":1}, {"label":"R", "x":4.5, "y":1}, {"label":"T", "x":5.5, "y":1}, {"label":"Y", "x":6.5, "y":1}, {"label":"U", "x":7.5, "y":1}, {"label":"I", "x":8.5, "y":1}, {"label":"O", "x":9.5, "y":1}, {"label":"P", "x":10.5, "y":1}, {"label":"{", "x":11.5, "y":1}, {"label":"}", "x":12.5, "y":1}, {"label":"Caps Lock", "x":0, "y":2, "w":1.75}, {"label":"A", "x":1.75, "y":2}, {"label":"S", "x":2.75, "y":2}, {"label":"D", "x":3.75, "y":2}, {"label":"F", "x":4.75, "y":2}, {"label":"G", "x":5.75, "y":2}, {"label":"H", "x":6.75, "y":2}, {"label":"J", "x":7.75, "y":2}, {"label":"K", "x":8.75, "y":2}, {"label":"L", "x":9.75, "y":2}, {"label":":", "x":10.75, "y":2}, {"label":"@", "x":11.75, "y":2}, {"label":"~", "x":12.75, "y":2}, {"label":"Enter", "x":13.75, "y":2, "w":1.25}, {"label":"Shift", "x":0, "y":3, "w":1.25}, {"label":"|", "x":1.25, "y":3}, {"label":"Z", "x":2.25, "y":3}, {"label":"X", "x":3.25, "y":3}, {"label":"C", "x":4.25, "y":3}, {"label":"V", "x":5.25, "y":3}, {"label":"B", "x":6.25, "y":3}, {"label":"N", "x":7.25, "y":3}, {"label":"M", "x":8.25, "y":3}, {"label":"<", "x":9.25, "y":3}, {"label":">", "x":10.25, "y":3}, {"label":"?", "x":11.25, "y":3}, {"label":"Shift", "x":12.25, "y":3, "w":2.75}, {"label":"Ctrl", "x":0, "y":4, "w":1.5}, {"label":"Win", "x":1.5, "y":4}, {"label":"Alt", "x":2.5, "y":4, "w":1.5}, {"x":4, "y":4, "w":7}, {"label":"AltGr", "x":11, "y":4, "w":1.5}, {"label":"Menu", "x":12.5, "y":4}, {"label":"Ctrl", "x":13.5, "y":4, "w":1.5}]
```

Configurator would place keys out of order without this change.

File: keyboards/gskt00/gskt00.h (hunk 9 above)
Comment (nachie) [reply keyboards/gskt00/gskt00.h]: Changed

File: keyboards/gskt00/info.json (hunk 10 above)
Comment (nachie) [reply keyboards/gskt00/info.json]: Done. For the future, is there a trick to generating these?

File: keyboards/gskt00/keymaps/default/keymap.c [hunk 11]
```diff
@@ -0,0 +1,100 @@
+#include QMK_KEYBOARD_H
+
+const uint16_t PROGMEM fn_actions[] = {};
```
[the remark this answers is not in the document]
Comment (nachie) [reply keyboards/gskt00/keymaps/default/keymap.c]: Removed the fn_actions

File: keyboards/gskt00/rules.mk [hunk 12]
```diff
@@ -0,0 +1,59 @@
...
+EXTRAKEY_ENABLE ?= yes	# Audio control and System control(+450)
+CONSOLE_ENABLE ?= yes	# Console for debug(+400)
+COMMAND_ENABLE ?= no    # Commands for debug and configuration
+SLEEP_LED_ENABLE ?= no  # Breathing sleep LED during USB suspend
+NKRO_ENABLE ?= yes		# USB Nkey Rollover - if this doesn't work, see here: https://github.com/tmk/tmk_keyboard/wiki/FAQ#nkro-doesnt-work
+BACKLIGHT_ENABLE ?= no  # Enable keyboard backlight functionality
+AUDIO_ENABLE ?= no
+RGBLIGHT_ENABLE ?= no
+
+LAYOUTS = 60_ansi
```
[the remark this answers is not in the document]
Comment (nachie) [reply keyboards/gskt00/rules.mk]: Added the tsangan layouts here instead.

File: keyboards/gskt00/rules.mk [hunk 13]
```diff
@@ -0,0 +1,59 @@
...
+
+# Boot Section Size in *bytes*
+OPT_DEFS += -DBOOTLOADER_SIZE=4096
+
+BOOTLOADER = atmel-dfu
+
+# Build Options
+#   comment out to disable the options.
+#
+BOOTMAGIC_ENABLE ?= yes	# Virtual DIP switch configuration(+1000)
```
[the remark this answers is not in the document]
Comment (nachie) [reply keyboards/gskt00/rules.mk]: Done, removed them.

File: keyboards/gskt00/info.json (hunk 10 above)
Comment (noroadsleft) [reply keyboards/gskt00/info.json]: In short, without manual correction, ISO layouts fall apart a bit when the KLE Raw Data is run through the QMK KLE-to-JSON Converter. It has to do with how the Converter assigns key order.

I work around this by building my ISO layouts [with a 1.25x1u Enter key on the home row](http://www.keyboard-layout-editor.com/##@@=%C2%AC%0A%60&=!%0A1&=%22%0A2&=%C2%A3%0A3&=$%0A4&=%25%0A5&=%5E%0A6&=%2F&%0A7&=*%0A8&=(%0A9&=)%0A0&=%2F_%0A-&=+%0A%2F=&_w:2%3B&=Backspace%3B&@_w:1.5%3B&=Tab&=Q&=W&=E&=R&=T&=Y&=U&=I&=O&=P&=%7B%0A%5B&=%7D%0A%5D%3B&@_w:1.75%3B&=Caps%20Lock&=A&=S&=D&=F&=G&=H&=J&=K&=L&=%2F:%0A%2F%3B&=%2F@%0A%27&=~%0A%23&_w:1.25%3B&=Enter%3B&@_w:1.25%3B&=Shift&=%7C%0A%5C&=Z&=X&=C&=V&=B&=N&=M&=%3C%0A,&=%3E%0A.&=%3F%0A%2F%2F&_w:2.75%3B&=Shift%3B&@_w:1.25%3B&=Ctrl&_w:1.25%3B&=Win&_w:1.25%3B&=Alt&_a:7&w:6.25%3B&=&_a:4&w:1.25%3B&=AltGr&_w:1.25%3B&=Win&_w:1.25%3B&=Menu&_w:1.25%3B&=Ctrl), running the data through the Converter, and then editing the resulting JSON manually.

I've done a lot of `info.json` files.

File: keyboards/gskt00/config.h [hunk 14]
```diff
@@ -0,0 +1,41 @@
...
+/* Locking resynchronize hack */
+#define LOCKING_RESYNC_ENABLE
+
+/* key combination for command */
+#define IS_COMMAND() ( \
+    keyboard_report->mods == (MOD_BIT(KC_LSHIFT) | MOD_BIT(KC_RSHIFT)) \
+)
+
+/* prevent stuck modifiers */
+#define PREVENT_STUCK_MODIFIERS
```
Comment (drashna) [inline keyboards/gskt00/config.h]: Could you remove this?  

We just made this the default, so this line is no longer needed (or useful)

Comment (nachie) [reply keyboards/gskt00/config.h]: Removed.

**Comment by drashna** [comment]
Thanks!

**Pull request merged**